core/src/main/java/hudson/Launcher.java

```diff
@@ -1298,16 +1299,20 @@ public RemoteProcess call() throws IOException {
 
             final Proc p = ps.start();
 
-            return Channel.current().export(RemoteProcess.class,new RemoteProcess() {
+            return channel.export(RemoteProcess.class,new RemoteProcess() {
                 public int join() throws InterruptedException, IOException {
                     try {
                         return p.join();
                     } finally {
                         // make sure I/O is delivered to the remote before we return
+                        Channel taskChannel = null;
                         try {
-                            Channel.current().syncIO();
+                            // Sync IO will fail automatically if the channel is being closed
+                            taskChannel = Channel.currentOrFail();
+                            taskChannel.syncIO();
```

Comment:
Given the comment above, why is this change needed at all?

Why the variable? Just to better debug?

Reply:
It is explanation why I do not use `getOpenChannelOrFail()`, will update the comment